mp/src/game/shared/gamemovement.cpp

```diff
@@ -3931,26 +3931,34 @@ void CGameMovement::CheckFalling( void )
 	player->m_Local.m_flFallVelocity = 0.0f;
 }
 
-void CGameMovement::PlayerRoughLandingEffects( float fvol )
+void CGameMovement::PlayerRoughLandingEffects(float fvol, short surfacePropsIndex)
 {
+	surfacedata_t* surfaceData;
+
 	if ( fvol > 0.0 )
 	{
-		//
 		// Play landing sound right away.
 		player->m_flStepSoundTime = 400;
 
-		// Play step sound for current texture.
-		player->PlayStepSound( (Vector &)mv->GetAbsOrigin(), player->m_pSurfaceData, fvol, true );
-
-		//
-		// Knock the screen around a little bit, temporary effect.
-		//
-		player->m_Local.m_vecPunchAngle.Set( ROLL, player->m_Local.m_flFallVelocity * 0.013 );
-
-		if ( player->m_Local.m_vecPunchAngle[PITCH] > 8 )
+		if (surfacePropsIndex == -1)
 		{
-			player->m_Local.m_vecPunchAngle.Set( PITCH, 8 );
+			// Play step sound for the surface the player is standing on if no
+			// override is specified.
+			surfaceData = player->m_pSurfaceData;
 		}
+		else
+		{
+			// Play step sound for specified surface.
+			surfaceData = MoveHelper()->GetSurfaceProps()->GetSurfaceData(surfacePropsIndex);
+
+			// Fallback in case of bad index.
+			if (!surfaceData)
+			{
+				surfaceData = player->m_pSurfaceData;
+			}
+		}
```

Comment:
This can be simplified as well:

```cpp
player->m_flStepSoundTime = 400;

surfacedata_t *pSurfaceData = nullptr;
if (surfacePropsIndexOverride != -1)
{
    pSurfaceData = MoveHelper()->GetSurfaceProps()->GetSurfaceData(surfacePropsIndexOverride);
}

if (!pSurfaceData)
{
    pSurfaceData = player->m_pSurfaceData;
}

pPlayer->PlayStepSound( ... );
```

Comments are really not needed; clean code should be very clear as to what it's doing, so if it's not, try to use better naming.